### PR TITLE
applying naming conventions

### LINKED ISSUE #32: Come with a naming convention for Benchpress cmdlets
Currently benchpress helper functions are named like, Get-XXX, Get-XXXExists

Should we add Benchpress or part of it in the naming, like, Get-BenchXXX, etc?

_Acceptance Criteria_

- [ ] Decide a naming schema
- [ ] Update helpers according to the schema

### PR DESCRIPTION
**Pull request by polatengin**
closes #32

as discussed in the design session,

`AzBP` prefix added to all helper cmdlets
`Exists` keyword replaced with `Exist` keyword

also, to be compliant with Powershell naming guidelines, I converted camel casing to pascal casing for cmdlet parameters

PS : noticed that I missed adding couple of nested modules and their functions list in BenchPress.Azure module. fixed that, too.

### CODE REVIEW

File: BenchPress/Helpers/Azure/AKSCluster.psm1 [hunk 1]
```diff
@@ -1,31 +1,46 @@
...
   )
 
-  $resource = Get-AzAksCluster -ResourceGroupName $resourceGroupName -Name $aksName
+  $resource = Get-AzAksCluster -ResourceGroupName $ResourceGroupName -Name $AKSName
   Write-Host $resource
   return $resource
 }
 
-function Get-AKSClusterExists {
+function Get-AKSCluster{
```
Comment (msha1026) [inline BenchPress/Helpers/Azure/AKSCluster.psm1:32]: Missing "Exist"

File: BenchPress/Helpers/Azure/ActionGroup.psm1 [hunk 2]
```diff
@@ -1,30 +1,61 @@
...
+.PARAMETER ActionGroupName
+  The name of the Azure Action Group
+
+.PARAMETER ResourceGroupName
+  The name of the resource group
+
+.EXAMPLE
+  Get-ActionGroup-ActionGroupName "benchpresstest" -ResourceGroupName "rgbenchpresstest"
+#>
+function Get-ActionGroup{
```
Comment (msha1026) [inline BenchPress/Helpers/Azure/ActionGroup.psm1:47]: Missing "Exist"

File: BenchPress/Helpers/Azure/ActionGroup.psm1 [hunk 3]
```diff
@@ -1,30 +1,61 @@
...
   )
 
-  $resource = Get-ActionGroup $actionGroupName $resourceGroupName
+  $resource = Get-ActionGroup -ActionGroupName $actionGroupName -ResourceGroupName $ResourceGroupName
   return ($null -ne $resource)
 }
 
-Export-ModuleMember -Function `
-  Get-ActionGroup, Get-ActionGroupExists
+Export-ModuleMember -Function Get-ActionGroup, Get-ActionGroupExists
```
Comment (msha1026) [inline BenchPress/Helpers/Azure/ActionGroup.psm1:61]: Change Exists -> Exist

File: BenchPress/Helpers/Azure/ContainerRegistry.psm1 [hunk 4]
```diff
@@ -8,11 +24,27 @@ function Get-ContainerRegistry {
...
+.PARAMETER Name
+  The name of the Container Registry
+
+.PARAMETER ResourceGroupName
+  The name of the resource group
+
+.EXAMPLE
+  Get-ContainerRegistry-Name "benchpresstest" -ResourceGroupName "rgbenchpresstest"
+#>
+function Get-ContainerRegistry{
```
Comment (msha1026) [inline BenchPress/Helpers/Azure/ContainerRegistry.psm1:47]: Missing "Exist"

File: BenchPress/Helpers/Azure/ContainerRegistry.psm1 [hunk 5]
```diff
@@ -21,9 +53,8 @@ function Get-ContainerRegistryExists {
...
     [string]$ResourceGroupName
   )
-  $resource = Get-ContainerRegistry -Name $Name -ResourceGroupName $resourceGroupName
+  $resource = Get-ContainerRegistry -Name $Name -ResourceGroupName $ResourceGroupName
   return ($null -ne $resource)
 }
 
-Export-ModuleMember -Function `
-  Get-ContainerRegistry, Get-ContainerRegistryExists
+Export-ModuleMember -Function Get-ContainerRegistry, Get-ContainerRegistryExists
```
Comment (msha1026) [inline BenchPress/Helpers/Azure/ContainerRegistry.psm1:60]: Exists -> Exist

File: BenchPress/Helpers/Azure/KeyVault.psm1 [hunk 6]
```diff
@@ -35,25 +35,26 @@ Helper function for KeyVault
...
-.PARAMETER resourceGroupName
+.PARAMETER ResourceGroupName
   The name of the resource group
 
 .EXAMPLE
-  Get-KeyVaultExists -name "kvbenchpresstest" -resourceGroupName "rgbenchpresstest"
+  Get-KeyVault-Name "kvbenchpresstest" -ResourceGroupName "rgbenchpresstest"
 #>
-function Get-KeyVaultExists {
+function Get-KeyVault{
```
Comment (msha1026) [inline BenchPress/Helpers/Azure/KeyVault.psm1:47]: Missing "Exist"

File: BenchPress/Helpers/Azure/KeyVault.psm1 [hunk 7]
```diff
@@ -35,25 +35,26 @@ Helper function for KeyVault
...
+.PARAMETER Name
   The name of the KeyVault
 
-.PARAMETER resourceGroupName
+.PARAMETER ResourceGroupName
   The name of the resource group
 
 .EXAMPLE
-  Get-KeyVaultExists -name "kvbenchpresstest" -resourceGroupName "rgbenchpresstest"
+  Get-KeyVault-Name "kvbenchpresstest" -ResourceGroupName "rgbenchpresstest"
```
Comment (msha1026) [inline BenchPress/Helpers/Azure/KeyVault.psm1:45]: Missing space between KeyVault and -Name

File: BenchPress/Helpers/Azure/ContainerRegistry.psm1 [hunk 8]
```diff
@@ -8,11 +24,27 @@ function Get-ContainerRegistry {
...
+  Helper function for Container Registry
+
+.PARAMETER Name
+  The name of the Container Registry
+
+.PARAMETER ResourceGroupName
+  The name of the resource group
+
+.EXAMPLE
+  Get-ContainerRegistry-Name "benchpresstest" -ResourceGroupName "rgbenchpresstest"
```
Comment (msha1026) [inline BenchPress/Helpers/Azure/ContainerRegistry.psm1:45]: Missing space between Get-ContainerRegistry and -Name

File: BenchPress/Helpers/Azure/ActionGroup.psm1 [hunk 9]
```diff
@@ -1,30 +1,61 @@
...
+  Helper function for Azure Action Group
+
+.PARAMETER ActionGroupName
+  The name of the Azure Action Group
+
+.PARAMETER ResourceGroupName
+  The name of the resource group
+
+.EXAMPLE
+  Get-ActionGroup-ActionGroupName "benchpresstest" -ResourceGroupName "rgbenchpresstest"
```
Comment (msha1026) [inline BenchPress/Helpers/Azure/ActionGroup.psm1:45]: Missing space between `ActionGroup` and `-ActionGroupName`

File: BenchPress/Helpers/Azure/AKSCluster.psm1 [hunk 10]
```diff
@@ -1,31 +1,46 @@
...
+  Helper function for AKS Cluster
+
+.PARAMETER AKSName
+  The name of the AKS Cluster
+
+.PARAMETER ResourceGroupName
+  The name of the resource group
+
+.EXAMPLE
+  Get-AKSCluster -ActionGroupName "benchpresstest" -ResourceGroupName "rgbenchpresstest"
```
Comment (msha1026) [inline BenchPress/Helpers/Azure/AKSCluster.psm1:15]: `-ActionGroupName` -> `AKSName`

File: BenchPress/Helpers/Azure/KeyVault.psm1 [hunk 11]
```diff
@@ -93,25 +95,26 @@ function Get-KeyVaultSecret {
...
+.PARAMETER Name
   The name of the KeyVault Secret
 
-.PARAMETER keyVaultName
+.PARAMETER KeyVaultName
   The name of the KeyVault
 
 .EXAMPLE
-  Get-KeyVaultSecretExists -name "samplesecret" -keyVaultName "kvbenchpresstest"
+  Get-KeyVaultSecret-Name "samplesecret" -KeyVaultName "kvbenchpresstest"
```
Comment (msha1026) [inline BenchPress/Helpers/Azure/KeyVault.psm1:105]: `KeyVaultSecret` -> `KeyVaultSecretExist` and missing space before `-Name`

File: BenchPress/Helpers/Azure/KeyVault.psm1 [hunk 12]
```diff
@@ -93,25 +95,26 @@ function Get-KeyVaultSecret {
...
 #>
-function Get-KeyVaultSecretExists {
+function Get-KeyVaultSecret{
```
Comment (msha1026) [inline BenchPress/Helpers/Azure/KeyVault.psm1:107]: Missing `Exist`

File: BenchPress/Helpers/Azure/KeyVault.psm1 [hunk 13]
```diff
@@ -151,25 +155,26 @@ function Get-KeyVaultKey {
...
+.PARAMETER Name
   The name of the KeyVault Key
 
-.PARAMETER keyVaultName
+.PARAMETER KeyVaultName
   The name of the KeyVault
 
 .EXAMPLE
-  Get-KeyVaultKeyExists -name "samplekey" -keyVaultName "kvbenchpresstest"
+  Get-KeyVaultKey-Name "samplekey" -KeyVaultName "kvbenchpresstest"
```
Comment (msha1026) [inline BenchPress/Helpers/Azure/KeyVault.psm1:165]: `KeyVaultKey` -> `KeyVaultKeyExist` and missing space before -Name

File: BenchPress/Helpers/Azure/KeyVault.psm1 [hunk 14]
```diff
@@ -151,25 +155,26 @@ function Get-KeyVaultKey {
...
 #>
-function Get-KeyVaultKeyExists {
+function Get-KeyVaultKey{
```
Comment (msha1026) [inline BenchPress/Helpers/Azure/KeyVault.psm1:167]: Missing `Exist`

File: BenchPress/Helpers/Azure/KeyVault.psm1 [hunk 15]
```diff
@@ -209,25 +215,26 @@ function Get-KeyVaultCertificate {
...
+.PARAMETER Name
   The name of the KeyVault Certificate
 
-.PARAMETER keyVaultName
+.PARAMETER KeyVaultName
   The name of the KeyVault
 
 .EXAMPLE
-  Get-KeyVaultCertificateExists -name "samplecertificate" -keyVaultName "kvbenchpresstest"
+  Get-KeyVaultCertificate-Name "samplecertificate" -KeyVaultName "kvbenchpresstest"
```
Comment (msha1026) [inline BenchPress/Helpers/Azure/KeyVault.psm1:225]: `KeyVaultCertificate` -> `KeyVaultCertificateExist` and missing space before `-Name`

File: BenchPress/Helpers/Azure/ResourceGroup.psm1 [hunk 16]
```diff
@@ -1,24 +1,49 @@
...
+  Helper function for Resource Group
+
+.DESCRIPTION
+  Helper function for Resource Group
+
+.PARAMETER ResourceGroupName
+  The name of the resource group
+
+.EXAMPLE
+  Get-ResourceGroup-ResourceGroupName "rgbenchpresstest"
```
Comment (msha1026) [inline BenchPress/Helpers/Azure/ResourceGroup.psm1:36]: `ResourceGroup` -> `ResourceGroupExist` and missing space before `-ResourceGroupName`

File: BenchPress/Helpers/Azure/ResourceGroup.psm1 [hunk 17]
```diff
@@ -1,24 +1,49 @@
...
   )
 
-  $resource = Get-ResourceGroup $resourceGroupName
+  $resource = Get-ResourceGroup $ResourceGroupName
   return ($null -ne $resource)
 }
 
-Export-ModuleMember -Function `
-  Get-ResourceGroup, Get-ResourceGroupExists
+Export-ModuleMember -Function Get-ResourceGroup, Get-ResourceGroupExists
```
Comment (msha1026) [inline BenchPress/Helpers/Azure/ResourceGroup.psm1:49]: `Exists` -> `Exist`

File: BenchPress/Helpers/Azure/KeyVault.psm1 [hunk 18]
```diff
@@ -209,25 +215,26 @@ function Get-KeyVaultCertificate {
...
 
     [Parameter(Mandatory = $true)]
-    [string]$keyVaultName
+    [string]$KeyVaultName
   )
-  $resource = Get-KeyVaultCertificate $name $keyVaultName
+
+  $resource = Get-KeyVaultCertificate -Name $Name -KeyVaultName $KeyVaultName
   return ($null -ne $resource)
 }
```
Comment (msha1026) [inline BenchPress/Helpers/Azure/KeyVault.psm1:240]: Edit exported cmdlets to change `Exists` -> `Exist`

File: BenchPress/Helpers/Azure/ServicePlan.psm1 [hunk 19]
```diff
@@ -1,29 +1,61 @@
...
+  Helper function for App Service Plan
+
+.PARAMETER AppServicePlanName
+  The name of the App Service Plan
+
+.PARAMETER ResourceGroupName
+  The name of the resource group
+
+.EXAMPLE
+  Get-AppServicePlan-AppServicePlanName "appsvcbenchpresstest" -ResourceGroupName "rgbenchpresstest"
```
Comment (msha1026) [inline BenchPress/Helpers/Azure/ServicePlan.psm1:45]: `AppServicePlan` -> `AppServicePlanExist` and missing space before `-AppServicePlanName`

File: BenchPress/Helpers/Azure/ServicePlan.psm1 [hunk 20]
```diff
@@ -1,29 +1,61 @@
...
+#>
+function Get-AppServicePlan{
```
Comment (msha1026) [inline BenchPress/Helpers/Azure/ServicePlan.psm1:47]: Missing `Exist`

File: BenchPress/Helpers/Azure/ServicePlan.psm1 [hunk 21]
```diff
@@ -1,29 +1,61 @@
...
   )
-  $resource = Get-AppServicePlan $appServicePlanName $resourceGroupName
+
+  $resource = Get-AppServicePlan -AppServicePlanName $AppServicePlanName -ResourceGroupName $ResourceGroupName
   return ($null -ne $resource)
 }
 
-Export-ModuleMember -Function `
-  Get-AppServicePlan, Get-AppServicePlanExists
+Export-ModuleMember -Function Get-AppServicePlan, Get-AppServicePlanExists
```
Comment (msha1026) [inline BenchPress/Helpers/Azure/ServicePlan.psm1:61]: `Exists` -> `Exist`

File: BenchPress/Helpers/Azure/SqlDatabase.psm1 [hunk 22]
```diff
@@ -2,34 +2,34 @@ function Get-SqlDatabase {
...
+    [string]$ResourceGroupName
   )
 
-  $resource = Get-AzSqlDatabase -ResourceGroupName $resourceGroupName -ServerName $serverName -DatabaseName $databaseName
+  $resource = Get-AzSqlDatabase -ResourceGroupName $ResourceGroupName -ServerName $ServerName -DatabaseName $DatabaseName
   return $resource
 }
 
-function Get-SqlDatabaseExists {
+function Get-SqlDatabase{
```
Comment (msha1026) [inline BenchPress/Helpers/Azure/SqlDatabase.psm1:18]: Missing `Exist`

File: BenchPress/Helpers/Azure/SqlDatabase.psm1 [hunk 23]
```diff
@@ -2,34 +2,34 @@ function Get-SqlDatabase {
...
   )
-  $resource = Get-SqlDatabase $databaseName $serverName $resourceGroupName
+
+  $resource = Get-SqlDatabase -DatabaseName $DatabaseName -ServerName $ServerName -ResourceGroupName $ResourceGroupName
   return ($null -ne $resource)
 }
 
-Export-ModuleMember -Function `
-  Get-SqlDatabase, Get-SqlDatabaseExists
+Export-ModuleMember -Function Get-SqlDatabase, Get-SqlDatabaseExists
```
Comment (msha1026) [inline BenchPress/Helpers/Azure/SqlDatabase.psm1:35]: `Exists` -> `Exist`

File: BenchPress/Helpers/Azure/SqlServer.psm1 [hunk 24]
```diff
@@ -2,29 +2,28 @@ function Get-SqlServer {
...
+    [string]$ResourceGroupName
   )
 
-  $resource = Get-AzSqlServer -ResourceGroupName $resourceGroupName -ServerName $serverName
+  $resource = Get-AzSqlServer -ResourceGroupName $ResourceGroupName -ServerName $ServerName
   return $resource
 }
 
-function Get-SqlServerExists {
+function Get-SqlServer{
```
Comment (msha1026) [inline BenchPress/Helpers/Azure/SqlServer.psm1:15]: Missing `Exist`

File: BenchPress/Helpers/Azure/SqlServer.psm1 [hunk 25]
```diff
@@ -2,29 +2,28 @@ function Get-SqlServer {
...
   )
 
-  $resource = Get-SqlServer $serverName $resourceGroupName
+  $resource = Get-SqlServer -ServerName $ServerName -ResourceGroupName $ResourceGroupName
   return ($null -ne $resource)
 }
 
-Export-ModuleMember -Function `
-  Get-SqlServer, Get-SqlServerExists
+Export-ModuleMember -Function Get-SqlServer, Get-SqlServerExists
```
Comment (msha1026) [inline BenchPress/Helpers/Azure/SqlServer.psm1:29]: `Exists` -> `Exist`

File: BenchPress/Helpers/Azure/VirtualMachine.psm1 [hunk 26]
```diff
@@ -2,29 +2,28 @@ function Get-VirtualMachine {
...
+    [string]$ResourceGroupName
   )
 
-  $resource = Get-AzVM -ResourceGroupName $resourceGroupName -Name $virtualMachineName
+  $resource = Get-AzVM -ResourceGroupName $ResourceGroupName -Name $VirtualMachineName
   return $resource
 }
 
-function Get-VirtualMachineExists {
+function Get-VirtualMachine{
```
Comment (msha1026) [inline BenchPress/Helpers/Azure/VirtualMachine.psm1:15]: Missing `Exist`

File: BenchPress/Helpers/Azure/VirtualMachine.psm1 [hunk 27]
```diff
@@ -2,29 +2,28 @@ function Get-VirtualMachine {
...
   )
 
-  $resource = Get-VirtualMachine $virtualMachineName $resourceGroupName
+  $resource = Get-VirtualMachine -VirtualMachineName $VirtualMachineName -ResourceGroupName $ResourceGroupName
   return ($null -ne $resource)
 }
 
-Export-ModuleMember -Function `
-  Get-VirtualMachine, Get-VirtualMachineExists
+Export-ModuleMember -Function Get-VirtualMachine, Get-VirtualMachineExists
```
Comment (msha1026) [inline BenchPress/Helpers/Azure/VirtualMachine.psm1:29]: `Exists` -> `Exist`

File: BenchPress/Helpers/Azure/WebApp.psm1 [hunk 28]
```diff
@@ -2,29 +2,28 @@ function Get-WebApp {
...
+    [string]$ResourceGroupName
   )
 
-  $resource = Get-AzWebApp -ResourceGroupName $resourceGroupName -Name $webAppName
+  $resource = Get-AzWebApp -ResourceGroupName $ResourceGroupName -Name $WebAppName
   return $resource
 }
 
-function Get-WebAppExists {
+function Get-WebApp{
```
Comment (msha1026) [inline BenchPress/Helpers/Azure/WebApp.psm1:15]: Missing `Exist`

File: BenchPress/Helpers/Azure/WebApp.psm1 [hunk 29]
```diff
@@ -2,29 +2,28 @@ function Get-WebApp {
...
   )
 
-  $resource = Get-WebApp $webAppName $resourceGroupName
+  $resource = Get-WebApp -WebAppName $WebAppName -ResourceGroupName $ResourceGroupName
   return ($null -ne $resource)
 }
 
-Export-ModuleMember -Function `
-  Get-WebApp, Get-WebAppExists
+Export-ModuleMember -Function Get-WebApp, Get-WebAppExists
```
Comment (msha1026) [inline BenchPress/Helpers/Azure/WebApp.psm1:29]: `Exists` -> `Exist`

**Review by msha1026 — CHANGES_REQUESTED** [review]

File: BenchPress/Helpers/Azure/AKSCluster.psm1 (hunk 1 above)
Comment (polatengin) [reply BenchPress/Helpers/Azure/AKSCluster.psm1]: 👍

File: BenchPress/Helpers/Azure/ActionGroup.psm1 (hunk 2 above)
Comment (polatengin) [reply BenchPress/Helpers/Azure/ActionGroup.psm1]: 👍

File: BenchPress/Helpers/Azure/ActionGroup.psm1 (hunk 3 above)
Comment (polatengin) [reply BenchPress/Helpers/Azure/ActionGroup.psm1]: 👍

File: BenchPress/Helpers/Azure/ContainerRegistry.psm1 (hunk 4 above)
Comment (polatengin) [reply BenchPress/Helpers/Azure/ContainerRegistry.psm1]: 👍

File: BenchPress/Helpers/Azure/KeyVault.psm1 (hunk 6 above)
Comment (polatengin) [reply BenchPress/Helpers/Azure/KeyVault.psm1]: 👍

File: BenchPress/Helpers/Azure/ContainerRegistry.psm1 (hunk 5 above)
Comment (polatengin) [reply BenchPress/Helpers/Azure/ContainerRegistry.psm1]: 👍

**Review by msha1026 — APPROVED** [review]
These changes will resolve the linting issues. I know we talked about removing Exist functions in general, but that is up to SA and PG. As long as long-term owners are onboard with these naming changes, then I think this is ok to merge!

**Pull request merged**